ibis/tests/all/test_aggregation.py

```diff
@@ -1,9 +1,73 @@
 import numpy as np
+import pandas as pd
 import pytest
 from pytest import param
 
 from ibis.tests.backends import Clickhouse, MySQL, Postgres, PySpark, SQLite
 
+aggregate_test_params = [
```

Comment:
Are the tests here different from what's defined here?

https://github.com/ibis-project/ibis/blob/master/ibis/tests/all/test_aggregation.py#L132

Reply:
They are a bit different

`test_aggregation` tests API methods that create `Reduction` nodes/ops (e.g. [`mean`](https://github.com/ibis-project/ibis/blob/13387c5a0969c99be403af5b3e8aa70bdb671623/ibis/expr/operations.py#L873), [`min`](https://github.com/ibis-project/ibis/blob/13387c5a0969c99be403af5b3e8aa70bdb671623/ibis/expr/operations.py#L960), [`max`](https://github.com/ibis-project/ibis/blob/13387c5a0969c99be403af5b3e8aa70bdb671623/ibis/expr/operations.py#L954)). Execution of these uses `aggcontext.agg` ([example](https://github.com/ibis-project/ibis/blob/master/ibis/pandas/execution/generic.py#L460)) and returns a scalar

However, the new `test_aggregate` and `test_aggregate_grouped` in this PR test the `aggregate` API method, which creates an `Aggregation` node/op. This has its own execution method and returns a dataframe

Reply:
Let me rename `test_aggregation` to `test_reduction_ops`. I think that will be clearer.

Reply:
Rename done

Reply:
I think what you have a pretty similar to;
https://github.com/ibis-project/ibis/blob/ec37e6e5ed14f4e7bc72dc2b70b07d85368f4376/ibis/tests/all/test_aggregation.py#L39

Can you try moving the new test to that file?

Reply:
i agree, happy to have you split out this to a separate file (moving all grouping ops tests)

Reply:
@icexelloss Sorry I am a bit confused, would you mind clarifying? The new tests are actually already in `test_aggregation.py`. Possibly there's a mix-up from above where I mentioned renaming `test_aggregation` to `test_reduction_ops`—that actually was to rename a specific test that was named `test_aggregation` to `test_reduction_ops` (not renaming the file named `test_aggregation.py`). Are you OK with splitting the file up like @jreback is suggesting?

This is what makes up [`test_aggregation.py`](https://github.com/ibis-project/ibis/blob/ec37e6e5ed14f4e7bc72dc2b70b07d85368f4376/ibis/tests/all/test_aggregation.py):

- Tests for `Aggregation` op (ungrouped) — new in this CR
- Tests for `Aggregation` op (grouped) — new in this CR
- Tests for `Reduction` subclass ops
- Tests for `TopK` op (I think `TopK` doesn't quite fit into either of the above. It's an aggregation/reduction that returns `K` number of rows, but isn't an `Aggregation` op nor a `Reduction` op)

I could split the file up this way:
1. The `Aggregation` op tests and `TopK` op tests in a file called `test_aggregation.py`
2. The `Reduction` op tests in a file called `test_reduction.py`

Reply:
Oh sorry I misread. Yes what you are doing now is what I suggested. So LGTM.

Reply:
lol, i see now as well.